File: lemniscate/src/test/java/com/vlad1m1r/lemniscate/utils/CurveUtilsTest.java
package com.vlad1m1r.lemniscate.utils;

import com.vlad1m1r.lemniscate.base.models.Point;

import org.junit.Test;

import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertSame;

/**
 * Created by vlad1m1r on 16-Oct-17.
 */
public class CurveUtilsTest {

    @Test
    public void checkPointForHole() {
        float viewSize = 100f;
        float strokeWidth = 10f;
        Point point = new Point(5,0, strokeWidth, viewSize);

        assertSame(CurveUtils.checkPointForHole(point, 1f, viewSize), point);
        assertNull(CurveUtils.checkPointForHole(point, 5f, viewSize));
        assertNull(CurveUtils.checkPointForHole(null, 1f, viewSize));
    }

}

File: lemniscate/src/test/java/com/vlad1m1r/lemniscate/utils/CurveUtilsTest.kt
package com.vlad1m1r.lemniscate.utils

import com.google.common.truth.Truth.assertThat
import com.vlad1m1r.lemniscate.base.models.Point
import org.junit.Test

class CurveUtilsTest {

    @Test
    fun checkPointForHole() {
        val point = Point(5f, 0f, 100f, 10f)

        assertThat(CurveUtils.checkPointForHole(point, 0.2f, 10f)).isSameAs(point)
        assertThat(CurveUtils.checkPointForHole(point, 5.0f, 10f)).isNull()
        assertThat(CurveUtils.checkPointForHole(null, 0.2f, 10f)).isNull()
    }
}